File: Components/WinForms/VB/GridTutorials/GridDoubleClickRow/GridDoubleClickRow.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Data
Imports DevExpress.XtraEditors.Repository

Namespace DevExpress.XtraGrid.Demos.Tutorials
    ''' <summary>
    ''' Summary description for GridDoubleClickRow.
    ''' </summary>
    Partial Public Class GridDoubleClickRow
        Inherits TutorialControl
        Public Sub New()
            '
            ' Required for Windows Form Designer support
            '
            InitializeComponent()

            '
            ' TODO: Add any constructor code after InitializeComponent call
            '
        End Sub

        Private Sub InitGrid()
            advBandedGridView1.OptionsBehavior.Editable = False
            advBandedGridView1.Columns("Transmiss Speed Count").GroupIndex = 0
            advBandedGridView1.ExpandAllGroups()
        End Sub

        Private Sub GridDoubleClickRow_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
            InitHelper.InitAdvBandedView(gridControl1)
            InitGrid()
        End Sub

        '<gridControl1>
        Private Sub gridControl1_DoubleClick(ByVal sender As Object, ByVal e As System.EventArgs) Handles gridControl1.DoubleClick
            Dim hi As DevExpress.XtraGrid.Views.BandedGrid.ViewInfo.BandedGridHitInfo = advBandedGridView1.CalcHitInfo((TryCast(sender, Control)).PointToClient(Control.MousePosition))

            If hi.RowHandle >= 0 Then
                ShowDataRow(advBandedGridView1.GetDataRow(hi.RowHandle), "", Color.White)
            ElseIf advBandedGridView1.FocusedRowHandle >= 0 Then
                ShowDataRow(advBandedGridView1.GetDataRow(advBandedGridView1.FocusedRowHandle), "Focused Row: ", Color.Yellow)
            Else
                ShowDataRow(Nothing, "Empty...", Color.DeepSkyBlue)
            End If
        End Sub
        '</gridControl1>

        Private Sub ShowDataRow(ByVal dr As DataRow, ByVal fs As String, ByVal c As Color)
            Dim s As String = ""
            If Not dr Is Nothing Then
                Dim items As Object() = dr.ItemArray
                For Each o As Object In items
                    If s = "" Then
                        s = ("") + o.ToString()
                    Else
                        s = (s & "; ") + o.ToString()
                    End If
                Next o
            End If
            lbRow.BackColor = c
            lbRow.Text = fs & s
        End Sub
        Public Overrides ReadOnly Property SetNewWhatsThisPadding() As Boolean
            Get
                Return True
            End Get
        End Property
    End Class
End Namespace

File: Components/WinForms/CS/GridTutorials/GridDoubleClickRow/GridDoubleClickRow.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using DevExpress.XtraEditors.Repository;

namespace DevExpress.XtraGrid.Demos.Tutorials {
	/// <summary>
	/// Summary description for GridDoubleClickRow.
	/// </summary>
	public partial class GridDoubleClickRow : TutorialControl {
		public GridDoubleClickRow() {
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

		private void InitGrid() {
			advBandedGridView1.OptionsBehavior.Editable = false;
			advBandedGridView1.Columns["Transmiss Speed Count"].GroupIndex = 0;
			advBandedGridView1.ExpandAllGroups();
		}

		private void GridDoubleClickRow_Load(object sender, System.EventArgs e) {
			InitHelper.InitAdvBandedView(gridControl1);
			InitGrid();
		}

      //<gridControl1>
		private void gridControl1_DoubleClick(object sender, System.EventArgs e) {
			DevExpress.XtraGrid.Views.BandedGrid.ViewInfo.BandedGridHitInfo hi = 
				advBandedGridView1.CalcHitInfo((sender as Control).PointToClient(Control.MousePosition));
			
			if(hi.RowHandle >= 0) 
				ShowDataRow(advBandedGridView1.GetDataRow(hi.RowHandle), "", Color.White);
			else if(advBandedGridView1.FocusedRowHandle >= 0)
                ShowDataRow(advBandedGridView1.GetFocusedDataRow(), "Focused Row: ", Color.Yellow);
			else
				ShowDataRow(null, "Empty...", Color.DeepSkyBlue);
		}
      //</gridControl1>

		private void ShowDataRow(DataRow dr, string fs, Color c) {
			string s = "";
			if(dr != null) {
				object[] items = dr.ItemArray;
				foreach(object o in items) 
					s = (s == "" ? "" : s + "; ") + o.ToString();
			}
			lbRow.BackColor = c;
			lbRow.Text = fs + s;
		}
        public override bool SetNewWhatsThisPadding { get { return true; } }
	}
}
